pyccel/parser/semantic.py

```diff
@@ -707,6 +706,13 @@ def _handle_function_args(self, arguments, **settings):
             if isinstance(a.value, StarredArguments):
                 args.extend([FunctionCallArgument(av) for av in a.value.args_var])
             else:
+                if isinstance(a.value, PyccelArithmeticOperator):
+                    d_var = self._infere_type(a.value)
+                    d_var.pop('datatype')
+                    var = self.scope.get_temporary_variable(a.value.dtype, **d_var)
+                    assi = Assign(var, a.value)
+                    self._additional_exprs[-1].append(assi)
+                    a = self._visit(FunctionCallArgument(var), **settings)
```

Comment:
Do you need to visit `FunctionCallArgument(var)`? The `var` object has a type/precision/rank/etc so doesn't it already have all necessary semantic behaviour?

Reply:
i removed the visit to `FunctionCallArgument`  in [51fe564](https://github.com/pyccel/pyccel/pull/1187/commits/51fe56464f3b9dcc1d309dcd559434a087da4b05).